openlibrary/plugins/importapi/code.py

```diff
@@ -338,24 +339,33 @@ def get_ia_record(metadata: dict) -> dict:
         lccn = metadata.get('lccn')
         subject = metadata.get('subject')
         oclc = metadata.get('oclc-id')
+        number_of_pages = metadata.get('imagecount')
         d = {
-            'title': metadata.get('title', ''),
+            'title': metadata.get('title', '').strip(),
             'authors': authors,
-            'publish_date': metadata.get('date'),
-            'publisher': metadata.get('publisher'),
+            'publish_date': metadata.get('date', '').strip(),
+            'publisher': metadata.get('publisher', '').strip(),
         }
         if description:
-            d['description'] = description
+            d['description'] = description.strip()
         if isbn:
             d['isbn'] = isbn
-        if language and len(language) == 3:
-            d['languages'] = [language]
+        if language:
+            if len(language) == 3:
+                d['languages'] = [language]
+
+            # Try converting the name of a language to its three character code.
+            # E.g. English -> eng.
+            elif lang_code := get_abbrev_from_full_lang_name(language):
+                d['languages'] = [lang_code]
         if lccn:
             d['lccn'] = [lccn]
         if subject:
             d['subjects'] = subject
         if oclc:
             d['oclc'] = oclc
+        if number_of_pages:
+            d['number_of_pages'] = int(number_of_pages)
```

Comment:
We might need to subtract 4 to get a better estimate of the number of pages, since the cover isn't usually included. Can you confirm that number makes sense?

```suggestion
            d['number_of_pages'] = int(number_of_pages)
```

Reply:
Of the 5 books I imported, the overestimate based on `imagecount` was:
- average: +8.6 pages
- median was +7
- minimum of +4
- maximum of +17

I will look into a bigger data set and report back with a recommendation and the data.

Reply:
I subtracted 4 from `imagecount` to get the page numbers, and ensured the page count wouldn't drop below 1.